### ISSUE TITLE
Issues with mixed-type parameter columns

### ISSUE THREAD

**Issue by asnyv** [equinor/webviz-subsurface#426]
Some users have design matrices with mixed types, e.g. a float and a string, where the latter identifies a case/scenario instead of a purely stochastic approach. 
This leads to a mixed-type `pandas` column (column type `object`) which can be problematic to compress and store as `parquet` files. [See e.g. this issue.](https://github.com/pandas-dev/pandas/issues/21228#issuecomment-462590774)

In addition, our parameter based plugins often expect numeric columns, and will filter out columns that aren't ([Example](https://github.com/equinor/webviz-subsurface/blob/3180bd1b99da862c9fe61e3be60d4725ad47ddcb/webviz_subsurface/plugins/_parameter_response_correlation.py#L248-L252)).

I therefore suggest that we don't take action to try to support these mixed-type columns, and rather promote the use of separate columns that work as flags for e.g. a scenario or a stochastic case, as these will generally be much easier to handle.
This also makes e.g. potential "global" parameter filters easier to implement in the future.

**Comment by perolavsvendsen** [comment]
Webviz is hardly the only thing that would struggle with mixed-type columns in Pandas, so sounds like a generic solution to expect homogenous-type (is that the term?) when data is given in e.g. Pandas.

But is the feedback to the user (error message, warnings, etc) of such a quality that it is understood, or does the user providing a mixed-type dataframe have to debug a lot to figure it out?

**Comment by asnyv** [comment]
> homogenous-type (is that the term?)

Actually think it is 👍 At least `pandas` documentation uses heterogeneous and homogeneous @perolavsvendsen 👏  

When it comes to current user feedback, the `parquet` issue currently leads to a traceback mainly from `pyarrow` which probably could be wrapped in a bit better in our case.

The plugins that assume numerical parameter columns, like the example above, quietly ignores columns that aren't, but (at least for most of them) it is stated in the documentation. Writing warnings for all ignored might raise more questions than it solves, as many projects e.g. have parameters that are only strings, and I don't see us going completely away from that anytime soon.
As seen from the **!Note** section if you scroll a bit [here](https://pandas.pydata.org/pandas-docs/stable/user_guide/basics.html#attributes-and-underlying-data), all columns that have at least one string entry will be of `dtype=object`, and if they e.g. are all strings then storing them as `parquet` is not an issue (even though the column is of type `dtype=object`).
Don't think it is feasible to loop through all values to check the individual types and verify homogeneity, so when it comes to ignoring columns, I we might just have to live with it (open for ideas of course).

**Comment by anders-kiaer** [comment]
> When it comes to current user feedback, the `parquet` issue currently leads to a traceback mainly from `pyarrow` which probably could be wrapped a bit better in our case.

Agree. For reference, here is a traceback example, raising error of type `pyarrow.lib.ArrowTypeError` which could perhaps be wrapped with a better error message in `webviz_store`:

```
  File "[...]/site-packages/webviz_config/webviz_store.py", line 182, in build_store
    output.to_parquet(f"{path}.parquet")
  File "[...]/site-packages/pandas/core/frame.py", line 2203, in to_parquet
    partition_cols=partition_cols, **kwargs)
  File "[...]/site-packages/pandas/io/parquet.py", line 252, in to_parquet
    partition_cols=partition_cols, **kwargs)
  File "[...]/site-packages/pandas/io/parquet.py", line 113, in write
    table = self.api.Table.from_pandas(df, **from_pandas_kwargs)
  File "pyarrow/table.pxi", line 1376, in pyarrow.lib.Table.from_pandas
  File "[...]/site-packages/pyarrow/pandas_compat.py", line 579, in dataframe_to_arrays
    for c, f in zip(columns_to_convert, convert_fields)]
  File "[...]/site-packages/pyarrow/pandas_compat.py", line 579, in <listcomp>
    for c, f in zip(columns_to_convert, convert_fields)]
  File "[...]/site-packages/pyarrow/pandas_compat.py", line 565, in convert_column
    raise e
  File "[...]/site-packages/pyarrow/pandas_compat.py", line 559, in convert_column
    result = pa.array(col, type=type_, from_pandas=True, safe=safe)
  File "pyarrow/array.pxi", line 265, in pyarrow.lib.array
  File "pyarrow/array.pxi", line 80, in pyarrow.lib._ndarray_to_array
  File "pyarrow/error.pxi", line 107, in pyarrow.lib.check_status
pyarrow.lib.ArrowTypeError: ("Expected a bytes object, got a 'float' object", 'Conversion failed for column SOME_PARAMETER_NAME with type object')
```

**Comment by anders-kiaer** [comment]
Transferring this issue to `webviz-config`, since we should not support mixed columns, and since a general, less technical error message, is to be added there.
